core/src/wallet/bitcoin/transaction_builders/BitcoinLikeStrategyUtxoPicker.cpp

```diff
@@ -503,17 +505,37 @@ namespace ledger {
             std::vector<bool> bestValues;
             int64_t bestValue = 0;
             buddy->logger->debug("Approximate Best Subset 1st try");
-            approximateBestSubset(vUTXOs, totalLower, BigInt((int64_t)amountWithFees), bestValues, bestValue);
-            if (bestValue != amountWithFees && totalLower >= amountWithFees + MIN_CHANGE) {
+            // Here we target the value amountWithFixedFees which is the amount of tx + fixed fees (fees of transaction without signed UTXOs)
+            approximateBestSubset(vUTXOs, totalLower, BigInt((int64_t)amountWithFixedFees), bestValues, bestValue);
+            if (bestValue != amountWithFixedFees && totalLower >= amountWithFixedFees + minimumChange) {
                 buddy->logger->debug("First approximation, bestValue {} with {} bestValues", bestValue, bestValues.size());
                 buddy->logger->debug("Approximate Best Subset 2nd try");
-                approximateBestSubset(vUTXOs, totalLower, BigInt((int64_t)amountWithFees + MIN_CHANGE), bestValues, bestValue);
+                approximateBestSubset(vUTXOs, totalLower, BigInt((int64_t)amountWithFixedFees + minimumChange), bestValues, bestValue);
                 buddy->logger->debug("Second approximation, bestValue {} with {} bestValues", bestValue, bestValues.size());
             }
 
+            // We compute the the totalBest here because we need picked UTXOs in subset
+            // to be able to update amountWithFees (because after signing all UTXOs we might need to pick additional UTXOs to cover
+            // fees generated by adding this UTXOs and their signature)
+            int64_t totalBest = 0;
+            UTXODescriptorList tmpOut;
+            for (unsigned int i = 0; i < vUTXOs.size(); i++) {
+                if (bestValues[i])
+                {
+                    UTXODescriptor utxoDescriptor{vUTXOs[i]->getTransactionHash(), vUTXOs[i]->getOutputIndex(), std::get<1>(buddy->request.utxoPicker.getValue())};
```

Comment:
Can you create a `constexpr` variable which represents the magic value `1` in the `std::get` ? Like that the code will be clearer

Reply:
I’m not quite sure what you mean here @alekece

Reply:
I mean `std::get<1>` is opaque and prone to error; if the underlying types in the tuple change - for some reason - this code will be break even if it looks great at the first glance. 
With that duck conversation with myself, my previous suggestion doesn't look that great : even if we put the magic value to a variable, the code will look better but the code is still error-proned.
Maybe a plain-old function to get back the value from that tuple will be less error-proned, more readable and reusable.

Reply:
Reusable? How so? Also, why do you want to make such a getter reusable? The problem here comes from the use of C++ tuples. It would be the same in another language, with a different syntax.

I think to either return a `struct`, or… just stick to that `get<1>` if it’s a tuple, which I think is okay for internal code.